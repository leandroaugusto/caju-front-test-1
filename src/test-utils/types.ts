export interface ITestUtils {
  children?: React.ReactNode
}
