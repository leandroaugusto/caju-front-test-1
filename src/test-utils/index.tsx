import { ReactElement } from "react";
import { ThemeProvider } from "styled-components";
import { render, RenderOptions, RenderResult } from "@testing-library/react";

import { ITestUtils } from "./types";

import theme from "~/theme";

const AllTheProviders: React.FC = ({ children }: ITestUtils) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
