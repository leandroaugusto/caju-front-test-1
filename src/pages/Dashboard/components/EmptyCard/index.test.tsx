import { customRender } from "~/test-utils";

import { EmptyCard } from ".";

describe("Columns", () => {
  it("Should render Columns component with correct data", () => {
    const { container } = customRender(<EmptyCard />);

    expect(container).toMatchSnapshot();
  });
});
