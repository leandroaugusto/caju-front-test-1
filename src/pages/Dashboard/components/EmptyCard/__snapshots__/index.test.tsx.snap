// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Columns > Should render Columns component with correct data 1`] = `
<div>
  <div
    class="sc-braxZu eYemEd"
    data-testid="empty-card-container"
  >
    <h3
      data-testid="employee-name"
    >
      NÃ£o foram encontrados registros para essa busca
    </h3>
  </div>
</div>
`;
